contracts/strategies/StrategyProxy.sol

```diff
@@ -113,7 +115,15 @@ contract StrategyProxy {
 
     function claim(address recipient) external {
         require(msg.sender == yveCRV, "!strategy");
-        uint256 amount = feeDistribution.claim(address(proxy));
+        if (now < lastTimeCursor) return;
+
+        while (feeDistribution.last_token_time() > feeDistribution.time_cursor_of(address(proxy))) {
```

Comment:
Would it make sense to limit the upper bound of this loop to avoid it going past the block gas limit?

Reply:
A transaction won't be accepted if it exceeds the block gas limit, but it does make sense to add a lower bound on a `while` loop, based on a loose estimate of gas left (leaving room for changes in the EVM, or make it configurable)

e.g. `while (gasLeft() > 10000 && ...)`

Reply:
I've modified the logic to avoid the looping..  if we're into a new epoch week it calls `claim_many` once, which amounts to 20 claims, and then records the new time position.  Looking at historic data it's typically less than 7 claims per week so I think this is a safe approach.